packages/twenty-server/src/modules/calendar/repositories/calendar-channel-event-association.repository.ts

```diff
@@ -169,4 +169,38 @@ export class CalendarChannelEventAssociationRepository {
       transactionManager,
     );
   }
+
+  public async deleteByCalendarEventParticipantHandleAndCalendarChannelIds(
+    calendarEventParticipantHandle: string,
+    calendarChannelIds: string[],
+    workspaceId: string,
+    transactionManager?: EntityManager,
+  ) {
+    const dataSourceSchema =
+      this.workspaceDataSourceService.getSchemaName(workspaceId);
+
+    const calendarChannelEventAssociationIdsToDelete =
+      await this.workspaceDataSourceService.executeRawQuery(
+        `SELECT "calendarChannelEventAssociation"."id"
+      FROM ${dataSourceSchema}."calendarChannelEventAssociation" "calendarChannelEventAssociation"
+      JOIN ${dataSourceSchema}."calendarEvent" "calendarEvent" ON "calendarChannelEventAssociation"."calendarEventId" = "calendarEvent"."id"
+      JOIN ${dataSourceSchema}."calendarEventParticipant" "calendarEventParticipant" ON "calendarEvent"."id" = "calendarEventParticipant"."calendarEventId"
+      WHERE "calendarEventParticipant"."handle" = $1 AND "calendarChannelEventAssociation"."calendarChannelId" = ANY($2)`,
+        [calendarEventParticipantHandle, calendarChannelIds],
+        workspaceId,
+        transactionManager,
+      );
+
+    const calendarChannelEventAssociationIdsToDeleteArray =
+      calendarChannelEventAssociationIdsToDelete.map(
+        (messageChannelMessageAssociation: { id: string }) =>
+          messageChannelMessageAssociation.id,
+      );
+
+    await this.deleteByIds(
```

Comment:
You can probably do a DELETE using directly your where clause above? This will avoid storing ids at this step when we don't need them